Userland/Applications/PartitionEditor/main.cpp

```diff
@@ -50,7 +51,10 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     window->resize(640, 400);
     window->set_icon(app_icon.bitmap_for_size(16));
 
-    // FIXME: Abort and show a dialog if not running as root.
+    if (getuid() != 0) {
+        GUI::MessageBox::show_error(window, "PartitionEditor must be run as root");
+        return Error::from_string_literal("PartitionEditor must be run as root");
```

Comment:
...but why?

Reply:
Block devices are owned by root and chmod 600. This is necessary to prevent unprivileged users from being able to access the block device directly and bypass filesystem permissions. To read partition tables, we need access to the raw block device though.

Reply:
Fine, but let's put a concise summary of that into the message and not just "root pls not telling you why tho"

Reply:
Changed to "PartitionEditor must be run as root in order to open raw block devices and read partition tables."